app/src/main/java/com/investrapp/investr/adapters/PlayerAdapter.java

```diff
@@ -0,0 +1,110 @@
+
+    private List<Player> mPlayers;
+    private Context mContext;
+    private Player currentPlayer;
+    private Competition competition;
+
+    class ViewHolder extends RecyclerView.ViewHolder {
+        private ImageView ivPlayerImage;
+        private TextView tvPlayerName;
+        private TextView tvPlayerDistance;
+        public ViewHolder(View itemView) {
+            super(itemView);
+            ivPlayerImage = (ImageView) itemView.findViewById(R.id.ivPlayerImage);
+            tvPlayerName = (TextView) itemView.findViewById(R.id.tvPlayerName);
+            tvPlayerDistance= (TextView) itemView.findViewById(R.id.tvPlayerDistance);
+        }
+    }
+
+    public PlayerAdapter(Context context, List<Player> players, Competition competition, Player currentPlayer) {
+        this.mContext = context;
+        this.mPlayers = players;
+        this.currentPlayer = currentPlayer;
+        this.competition = competition;
+    }
+
+    public Context getContext() {
+        return mContext;
+    }
+
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+        View vPlayers = inflater.inflate(R.layout.item_player, parent, false);
+        ViewHolder vhPlayers = new ViewHolder(vPlayers);
+        return vhPlayers;
+    }
+
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        final Player player = mPlayers.get(position);
+
+        TextView tvName = holder.tvPlayerName;
+        TextView tvDistance = holder.tvPlayerDistance;
+        ImageView ivProfilePicture = holder.ivPlayerImage;
+
+        Glide.with(getApplicationContext())
+                .load(player.getProfileImageUrl())
+                .centerCrop()
+                .into(ivProfilePicture);
+        tvName.setText(player.getName());
+
+        double otherPlayerLatitude = player.getLatitude();
+        double otherPlayerLongitude = player.getLongitude();
+
+
+        double currentPlayerLatitude = currentPlayer.getLatitude();
+        double currentPlayerLongitude = currentPlayer.getLongitude();
+
+        double distance = DistanceUtils.distFrom(currentPlayerLatitude, currentPlayerLongitude,
+                otherPlayerLatitude, otherPlayerLongitude);
+        //round the distance to one decimal place
+        distance = Math.round(distance * 10.0) / 10.0;
+        String distanceMessage = "" + distance + " miles away";
+        tvDistance.setText(distanceMessage);
+
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                CompetitionPlayer competitionPlayer = new CompetitionPlayer(competition, player);
+                ParseClient.addPlayerToCompetition(competitionPlayer);
+                String message = "Added " + player.getName() + " to " + competition.getName();
```

Comment:
@ this point, we also need to initialize the cash for the new player by inserting a cash transaction. See here for how I did it when a player decides to join a competition: https://github.com/InvestrApp/investrAndroid/blob/9795eca5a37128cfc982a1bb3d4ff47254b893e8/app/src/main/java/com/investrapp/investr/fragments/AllCompetitionsFragment.java#L91.

Reply:
Lines 91 - 101 in the above link.